Python/marshal.c

```diff
@@ -549,6 +554,10 @@ w_complex_object(PyObject *v, char flag, WFILE *p)
         Py_DECREF(pairs);
     }
     else if (PyCode_Check(v)) {
+        if (!p->allow_code) {
+            p->error = WFERR_UNMARSHALLABLE;
```

Comment:
Maybe it would be worth it to have a more specific error just for code, to have a more specific error message in PyMarshal_WriteObjectToString().